src/pykeen/models/base.py

```diff
@@ -183,6 +183,72 @@ def _process_remove_known(df: pd.DataFrame, remove_known: bool, testing: Optiona
     return df
 
 
+class RepresentationModule(nn.Module):
+    """A base class for obtaining representations for entities/relations."""
+
+    @property
+    def dimension(self) -> int:
+        """The representation dimension."""
+        raise NotImplementedError
+
+    @property
+    def total_size(self) -> int:
+        """The total number of representations (i.e. the maximum ID)"""
+        raise NotImplementedError
+
+    def forward(self, indices: torch.LongTensor) -> torch.FloatTensor:
+        """
+        Get representations for indices.
+
+        :param indices: shape: (m,)
+            The indices.
+
+        :return: shape: (m, d)
+            The representations.
+        """
+        raise NotImplementedError
+
+    def reset_parameters(self) -> None:
+        """Reset the module's parameters."""
+        pass
+
+
+class Embedding(RepresentationModule):
+    """Trainable embeddings."""
+
+    def __init__(
+        self,
+        num: int,
+        dim: int,
+        initialization: Callable[[nn.Parameter], None] = nn.init.normal_,
+        normalization: Optional[Callable[[torch.FloatTensor], torch.FloatTensor]] = None
+    ):
+        super().__init__()
+        self.initialization = initialization
+        self.normalization = normalization
+        self._embeddings = nn.Embedding(
+            num_embeddings=num,
+            embedding_dim=dim,
+        )
+
+    @property
+    def total_size(self) -> int:  # noqa: D102
```

Comment:
Why not name this num_embeddings to reflect the nn.Embedding?

Reply:
Maybe `max_id` would be better.

I think "representation" is a much broader term than "embedding", where the former one can be obtained from anything, e.g. via a CNN from an image, whereas the latter is usually a trained vector.